File: DataPath/Lua/CSharp2Lua/LuaCallStatic.lua
-- Lua调用静态类

-- 规则 CS.命名空间.类名.成员变量
print(CS.ZB.TestStatic.id);

-- 给静态属性赋值
CS.ZB.TestStatic.Name = "admin";
print(CS.ZB.TestStatic.Name)

-- 静态成员方法调用
-- 规则 CS.命名空间.类名.方法名
print(CS.ZB.TestStatic.Output());

CS.ZB.TestStatic.Default();
CS.ZB.TestStatic.Default("def");

File: Assets/Scripts/CSharp2Lua/LuaCallStatic.cs
using System;
using UnityEngine;

namespace ZB
{
    public static class TestStatic
    {
        public static int id = 99;
        
        public static string Name
        {
            get;
            set;
        }

        public static string Output()
        {
            return "Static Method";
        }

        public static void Default(string str = "abc")
        {
            Debug.Log(str);
        }
    }
}

public class LuaCallStatic : MonoBehaviour
{
    private void Start()
    {
        XLuaEnv.Instance.DoString("CSharp2Lua/LuaCallStatic");
    }

    private void OnDestroy()
    {
        XLuaEnv.Instance.Dispose();
    }
}